examples/loop_optimizations_service/service_py/loops_opt_service.py

```diff
@@ -176,7 +214,9 @@ def apply_action(self, action: Event) -> Tuple[bool, Optional[ActionSpace], bool
                 args[i] = arg
             run_command(
                 [
-                    "../opt_loops/opt_loops",
+                    runfiles_path(
+                        "examples/loop_optimizations_service/opt_loops/opt_loops"
+                    ),
```

Comment:
@ChrisCummins I have changed it to `runfiles_path` and it is working fine with bazel, but with CMake it is not working... I have added a CMakeLists.txt file to make `loops_opt_service.py` dependent on `opt_loops` but it is still not working. Do you have an idea how to get it working with CMake?

Reply:
Oof. Erm, I have no idea how data dependencies work in CMake, sorry! @sogartar?

Reply:
If the file exists, but resolving the path fails, then you could ditch `runfiles_path()` in favor of something like `Path(os.path.join(os.path.dirname(__file__), "path/to/my/thing")).resolve()` to get an absolute path to the dependency.

The only reason I suggested not using relpaths is in case the user `os.chdir()`s.

Reply:
The whole `runfiles_path()` mechanism will need a total overhaul when migrating fully to CMake build.

Reply:
Runfiles depends on Bazel if the environment variable `COMPILER_GYM_RUNFILES` is not defined. With CMake the Bazel resolution will probably fail, so it falls back to resolving the path as a relative path with respect to `_PACKAGE_ROOT`. Either you define the `COMPILER_GYM_RUNFILES` or have the file in the exact relative location with respect to `_PACKAGE_ROOT`.

Reply:
Another problem may be that the file does not get where it should be. This means that something is wrong with the dependencies and the target that makes the files is not triggered. There may be a bug in `cg_add_data_dependencies(...)`.
Another problem may come from `cg_genrule` resolving to the wrong command, but this should generate an error when executed.
@mostafaelhoushi, you can run the CMake configuration with `--trace-expand` flag and see what happens line-by-line.
You can also use CMake's `--graphviz` option to visualize the dependency graph.

Reply:
Thanks @sogartar for your guidance. I tried to debug it and I think this is the cause:
>> something is wrong with the dependencies and the target that makes the files is not triggered

For now I used the `Path(os.path.join(os.path.dirname(__file__), "path/to/my/thing")).resolve()` workaround and created this issue #603